rootfs/api/models/build.py

```diff
@@ -27,17 +27,28 @@ class Meta:
         ordering = ['-created']
         unique_together = (('app', 'uuid'),)
 
+    @property
+    def type(self):
+        """Figures out what kind of build type is being deal it with"""
+        if self.dockerfile:
+            return 'dockerfile'
+        elif self.sha:
+            return 'buildpack'
+        else:
+            # docker image (or any sort of image) used via deis pull
+            return 'image'
+
+    @property
+    def version(self):
+        return 'git-{}'.format(self.sha) if self.type == 'buildpack' else 'latest'
+
     def create(self, user, *args, **kwargs):
         latest_release = self.app.release_set.latest()
-        source_version = 'latest'
-        if self.sha:
-            source_version = 'git-{}'.format(self.sha)
-
         new_release = latest_release.new(
             user,
             build=self,
             config=latest_release.config,
-            source_version=source_version
+            source_version=self.version
```

Comment:
since source_version is just based on sha and not on the build type shouldn't version also be the same?